### ISSUE TITLE
Invalid calls reverted with `invalid opcode` instead of declared message

### ISSUE THREAD

**Issue by bitbeckers** [CommitPool/commitpool-contract-singleplayer#4]
**Issue**
In the contract we have multiple ```require``` checks to validate contract calls before execution. Currently, some invalid input returns and error with ```invalid opcode``` in the message instead of the expected message.

**Example**
line 174 SinglePlayerCommit.sol ``` require(!commitments[msg.sender].exists, "SPC::makeCommitment - msg.sender already has a commitment");```

line 111..116 SinglePlayerCommit.user.sol (test) 
```     
 //Activity
      //TODO improve to revertedWith; now returns invalid opcode instead of error message
      _activity = 'LALALA';

      await expect(
        contractWithUser.makeCommitment(_activity, _measureIndex, _goal, _startTime, _amountToStake, _overrides),
      ).to.be.reverted;
```

**Reference** 
https://solidity.readthedocs.io/en/v0.6.10/control-structures.html

**Expected outcome:**
1) When making a contract call with invalid input (for instance measureIndex out of range) the call is reverted with the error message declared in the contract.
2) Test cases for invalid input updated, TODO removed, checked with ```to.be.revertedWith```

**Comment by bitbeckers** [comment]
Resolved in https://github.com/CommitPool/commitpool-contract-singleplayer/tree/fix_console_log

Issues: invalid input ('LALALA'), now gave wrong address and getting expected message.

**Comment by bitbeckers** [comment]
Fixed with #10
